integrations/anthropic/src/haystack_integrations/components/generators/anthropic/chat/chat_generator.py

```diff
@@ -330,3 +377,104 @@ def _remove_cache_control(self, message: Dict[str, Any]) -> Dict[str, Any]:
         :returns: The message with the cache_control key removed.
         """
         return {k: v for k, v in message.items() if k != "cache_control"}
+
+    @component.output_types(replies=List[ChatMessage])
+    def run(
+        self,
+        messages: List[ChatMessage],
+        streaming_callback: Optional[Callable[[StreamingChunk], None]] = None,
+        generation_kwargs: Optional[Dict[str, Any]] = None,
+        tools: Optional[List[Tool]] = None,
+    ):
+        """
+        Invokes the Anthropic API with the given messages and generation kwargs.
+
+        :param messages: A list of ChatMessage instances representing the input messages.
+        :param streaming_callback: A callback function that is called when a new token is received from the stream.
+        :param generation_kwargs: Optional arguments to pass to the Anthropic generation endpoint.
+        :param tools: A list of tools for which the model can prepare calls. If set, it will override
+        the `tools` parameter set during component initialization.
+        :returns: A dictionary with the following keys:
+            - `replies`: The responses from the model
+        """
+        # update generation kwargs by merging with the generation kwargs passed to the run method
+        generation_kwargs = {**self.generation_kwargs, **(generation_kwargs or {})}
+        disallowed_params = set(generation_kwargs) - set(self.ALLOWED_PARAMS)
+        if disallowed_params:
+            logger.warning(
+                "Model parameters %s are not allowed and will be ignored. Allowed parameters are %s.",
+                disallowed_params,
+                self.ALLOWED_PARAMS,
+            )
+        generation_kwargs = {k: v for k, v in generation_kwargs.items() if k in self.ALLOWED_PARAMS}
+
+        system_messages, non_system_messages = _convert_messages_to_anthropic_format(messages)
+
+        # prompt caching
+        extra_headers = generation_kwargs.get("extra_headers", {})
+        prompt_caching_on = "anthropic-beta" in extra_headers and "prompt-caching" in extra_headers["anthropic-beta"]
```

Comment:
All looks good but we need to take care of this prompt caching headers as it is not in beta any more. But let's do that separately not to complicate things further